### ISSUE TITLE
LDAP RFC-4514 representation of distinguished names

### ISSUE THREAD

**Issue by trantor** [rrthomas/recode#33]
Hello @rrthomas .
I was interested in recode support for the peculiar encoding/escaping of the RFC in question.
Would it be within the scope of the software?

Thanks a lot

**Comment by rrthomas** [comment]
Yes, I think this would be within the scope of Recode. If you would like to implement it, I would be happy to discuss the design.

**Comment by trantor** [comment]
Hello @rrthomas and thanks for the answer.
Unfortunately I am not a C developer but I might attempt to code something in the near future, so I might, circumstances allowing so, take you up on that offer.
There are also a few other encoding conversions I've tried recently which seem wrong but I have to look harder to see whether it's something wrong in the code or in my understanding/expectations.
Forgive my bluntness, but judging from the commit history it seems to me that since taking over from the original creator (that passed away in 2014) the development of the software on your side has been almost entirely on cleaning/fixing/modernizing the code base, the building process and the localizations (i.e. mostly caretaking) rather than implementing something new when it comes to new encodings and such.
May I assume that further enhancements in that area, i.e. growing/enhancing the software in its primary purpose, would need to come from interested third parties?
Have a nice day :)

**Comment by rrthomas** [comment]
Hopefully the code to be written would be more-or-less trivial, and the conversion would be mostly data-driven.

You're quite right, my involvement with recode has been purely maintenance, to clean up and release François Pinard's unreleased development work, to integrate bug fixes, and to fix portability and other problems.

Anyone wanting to further develop recode would do well to read Pinard's own notes (the file `TODO`), as well as to consider whether another more modern framework might be a better place to do the work (for example, `iconv`).

Pinard himself planned to rewrite recode in Python; I would not now make that choice, but if I were planning recode's further development, I would certainly want to rewrite it in a more tractable language than C. Also, I would want to remove duplication by using iconv or other libraries for as many of the standard recodings as possible. (This would require careful auditing of the differences between recode and other systems.)
